src/components/importwms/ImportWmsDirective.js

```diff
@@ -186,6 +188,23 @@ goog.require('ga_urlutils_service');
               layer.Abstract = 'layer_invalid_no_name';
             }
 
+            // if there is no intersection between the map-extent
+            // and the WMS-layer-extent, the layer is set as invalid
+            if ((layer.Layer) && (layer.Layer !== undefined)) {
+              for (var i = 0; i < layer.Layer.length; i++) {
+                var l = getChildLayers(layer.Layer[i], map, wmsVersion);
+                if (l.extent) {
+                  var intersection =
+                    ol.extent.intersects(map.previousExtent_, l.extent);
+                }
+              }
+              if (!intersection) {
+                layer.isInvalid = true;
+                layer.Abstract =
+                  'layer_invalid_no_intersection_with_map_extent';
+              }
+            }
+
```

Comment:
`getChildLayers` is called recursively so you don't need to go through the childrens. That's why I suggest previously to test the intersections outside the loop.

Do only  : 

```
layer.extent = ol.extent.intersects(map.getView().getProjection().getExtent(), layer.extent);
if (!layer.extent) {
  layer.isInvalid = true;
  layer.Abstract =  'layer_invalid_no_intersection_with_map_extent';
}
```